webrtc/src/peer_connection/peer_connection_internal.rs

```diff
@@ -1358,30 +1356,27 @@ impl PeerConnectionInternal {
                 Some(r) => r,
                 None => continue,
             };
-            let mid = match transeiver.mid().await {
-                m if !m.is_empty() => m,
-                _ => continue,
-            };
-
-            let track = match sender.track().await {
-                Some(t) => t,
-                None => continue,
-            };
+            if let Some(mid) = transeiver.mid() {
```

Comment:
The old code was nice because it didn't increasing nesting, maybe keep it as

```rust
let mid = match transceiver.mid() {
  m if !m.is_empty() => m,
  _ => continue,
}
```

Strictly speaking `mid` shouldn't ever be empty any-more, maybe that's an invariant we should enforce when first setting it.

Alternatively `let-else` syntax

```rust
let Some(mid) = transceiver.mid() else {
  continue;
}
```